### ISSUE TITLE
False positive: duplicate assertion

### ISSUE THREAD

**Issue by SmirnovOleg** [JetBrains-Research/PyNose#41]
Project: pip
File link: https://github.com/pypa/pip/blob/12dad051804907fae1b2232fba3d218035881a60/tests/unit/test_req.py#L494

![image](https://user-images.githubusercontent.com/19374426/152305349-40dd60ce-7aff-43e4-8b2a-93e24564c3f0.png)

@OlesiaSub 
Do we need to check that there was a reassignment of the variable between duplicate assertions to avoid such FPs?

**Comment by OlesiaSub** [comment]
yes, I think we even discussed it a while ago. and I guess I could implement it.
